### DISCUSSION TITLE
kart türleri

### DISCUSSION THREAD

**Discussion by innthebox** [mewebstudio/pos#247] [Ideas]

**Comment by nuryagdym** [comment]
Selam,
tesekkurler, iyi fikir, musait olunca ekleriz.

**Comment by nuryagdym** [comment]
Bu ozelligi yeterli kadar oylama gelirse gelistiririz. Biraz arastirdim, kart firmalari yeni kart numaralari (patternleri) cikardikca bu ozellik sorun cikar. Bunun icin kutuphaneyi surekli guncel tutulmasi gerekecek. 
https://stackoverflow.com/a/55019607/4896948

Belki bunun icin baska kutuphane kullanilabilirsiniz.
Ornegin: https://github.com/jlorente/php-credit-cards

**Comment by zeydduran** [comment]
> Bu ozelligi yeterli kadar oylama gelirse gelistiririz. Biraz arastirdim, kart firmalari yeni kart numaralari (patternleri) cikardikca bu ozellik sorun cikar. Bunun icin kutuphaneyi surekli guncel tutulmasi gerekecek. https://stackoverflow.com/a/55019607/4896948
> 
> Belki bunun icin baska kutuphane kullanilabilirsiniz. Ornegin: https://github.com/jlorente/php-credit-cards

Bu soruna çözüm üretmek adına kart bin numaralarını veritabanında tutuyorum kullanıcılar (E-ticaret  sisteminin admini) bin numaralarını isteklerine göre düzenliyorlar. Örnek; Bin numarasına taksit,puan kullanma yetkisi, bin numarası hangi sanal pos ile çalışacak, bin numarası hangi organizasyona ait(visa,master vb.), bin numarası hangi bankaya ait, bin numarası hangi markaya ait (World,Bonus vb.), bin numara markası hangi sanal pos ile çalışacak.
Esnek bir yapı olduğunu düşünüyorum. 
Bankalar kendi müşterilerine bin numaralarında bir ekleme vs olduğunda mail ile bilgilendirme yapıyorlar. Bize bu tarz bilgilendirmeler yapmadıkları için böyle bir yapı kullanıyorum.

**Comment by Smokietr** [comment]
[https://lookup.binlist.net/](https://lookup.binlist.net/) üzerinden çekebilirsiniz. Dikkat limit var.

```

<?php

namespace App\Services;

use GuzzleHttp\Exception\GuzzleException;

class BinService
{
    protected \GuzzleHttp\Client $client;

    public function __construct()
    {
        $this->client = new \GuzzleHttp\Client([
            'verify' => false,
            'options' => [
                'CURLOPT_INTERFACE' => request()->ip // Set the IP address to be used for the request
            ]
        ]);
    }

    /**
     * @throws GuzzleException
     */
    private function getBinInformation(string $bin): array
    {
        $response = $this->client->get('https://lookup.binlist.net/' . $bin);

        if ($response->getStatusCode() !== 200) {
            return [];
        }

        return json_decode($response->getBody()->getContents(), true);
    }

    /**
     * @throws GuzzleException
     */
    public function getBinInformationByCardNumber(string $cardNumber): array
    {
        $bin = substr($cardNumber, 0, 6);

        return $this->getBinInformation($bin);
    }

    // TODO Exception handling
}
```
